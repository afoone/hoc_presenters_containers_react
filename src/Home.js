import React from 'react'

const HomePresenter = ({ someSideEffect }) => <div>{someSideEffect}</div>

const MemoizedHomePresenter = React.memo(HomePresenter)

// export const Home = () => {
//   const someSideEffect = () => 'mi resultado'

//   return <MemoizedHomePresenter someSideEffect={someSideEffect()} />
// }

const Home = ({ mockRender }) => {
  const someSideEffect = 'mi resultado'
  const renderComponent = text => (
    <MemoizedHomePresenter someSideEffect={text} />
  )
  const render = mockRender || renderComponent
  return render(someSideEffect)
}

export const _ = {
  HomePresenter,
  MemoizedHomePresenter,
}

export default Home
