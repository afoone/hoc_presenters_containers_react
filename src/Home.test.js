import { render, screen } from '@testing-library/react'
import React from 'react'
import Home, { _ } from './Home'

const { MemoizedHomePresenter } = _

describe('home', () => {
  it('should render', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })

  it('should render what i want', () => {
    render(<MemoizedHomePresenter someSideEffect='hello world' />)
    screen.getByText('hello world')
  })

  it('should pass props to the component', () => {
    const mockRender = jest.fn()
    render(<Home mockRender={mockRender} />)
    expect(mockRender).toHaveBeenCalledWith('mi resultado')
  })
})
