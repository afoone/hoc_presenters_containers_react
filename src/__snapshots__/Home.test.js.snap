// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should render 1`] = `
<div>
  <div>
    mi resultado
  </div>
</div>
`;
